### ISSUE TITLE
Wrong method signatures for `rem2pi` (causing failing tests)

### ISSUE THREAD

**Issue by rdeits** [JuliaDiff/ForwardDiff.jl#288]
The unit tests currently fail on master with:

```
ERROR: LoadError: LoadError: MethodError: no method matching rem2pi(::Int64, ::Float64)
Closest candidates are:
  rem2pi(::Int64, ::RoundingMode) at math.jl:903
  rem2pi(::ForwardDiff.Dual{Tx,V,N} where N where V<:Real, ::AbstractFloat) where Tx at /home/rdeits/Downloads/packages/v0.6/ForwardDiff/src/dual.jl:104
  rem2pi(::Integer, ::ForwardDiff.Dual{Ty,V,N} where N where V<:Real) where Ty at /home/rdeits/Downloads/packages/v0.6/ForwardDiff/src/dual.jl:105
  ...
Stacktrace:
 [1] rem2pi(::ForwardDiff.Dual{DualTest.TestTag(),Int64,1}, ::Float64) at /home/rdeits/Downloads/packages/v0.6/ForwardDiff/src/dual.jl:206
 [2] macro expansion at /home/rdeits/Downloads/packages/v0.6/ForwardDiff/test/DualTest.jl:421 [inlined]
 [3] anonymous at ./<missing>:?
 [4] include_from_node1(::String) at ./loading.jl:576
 [5] include(::String) at ./sysimg.jl:14
 [6] include_from_node1(::String) at ./loading.jl:576
 [7] include(::String) at ./sysimg.jl:14
 [8] process_options(::Base.JLOptions) at ./client.jl:305
 [9] _start() at ./client.jl:371
while loading /home/rdeits/Downloads/packages/v0.6/ForwardDiff/test/DualTest.jl, in expression starting on line 29
while loading /home/rdeits/Downloads/packages/v0.6/ForwardDiff/test/runtests.jl, in expression starting on line 10
```

There seem to be two issues here:

1. it looks like the unit test is trying to call `rem2pi` with two numeric arguments, which shouldn't work anyway
2. `rem2pi` seems to be treated as a generic binary numeric operation, so its `Dual` methods expect *both* arguments to be numbers or Duals, when in fact the second argument should be a rounding mode. 

Simple reproduction of (2):

```
julia> ForwardDiff.derivative(x -> rem2pi(x, RoundUp), 1.0)
ERROR: MethodError: no method matching rem2pi(::ForwardDiff.Dual{ForwardDiff.Tag{##3#4,Float64},Float64,1}, ::RoundingMode{:Up})
Closest candidates are:
  rem2pi(::Float64, ::RoundingMode{:Up}) at math.jl:870
  rem2pi(::Float32, ::RoundingMode) at math.jl:899
  rem2pi(::Float16, ::RoundingMode) at math.jl:900
  ...
Stacktrace:
 [1] derivative(::##3#4, ::Float64) at /home/rdeits/Downloads/packages/v0.6/ForwardDiff/src/derivative.jl:14
```

Perhaps `rem2pi` doesn't really fit the `DiffRules` model and should be special-cased in ForwardDiff?

@tkoolen any thoughts?

**Comment by tkoolen** [comment]
See https://github.com/JuliaDiff/DiffBase.jl/issues/18, https://github.com/JuliaDiff/DiffBase.jl/pull/19, especially https://github.com/JuliaDiff/DiffBase.jl/pull/19#issuecomment-330887197.

**Comment by rdeits** [comment]
Ah, gotcha. So it looks like the special case in ForwardDiff is still missing?

**Comment by jrevels** [comment]
fixed by #290
